paddle/framework/operator.h

```diff
@@ -262,6 +262,10 @@ class ExecutionContext : public InferShapeContext {
 
   platform::Place GetPlace() const { return device_context_->GetPlace(); }
 
+  const platform::DeviceContext* device_context() const {
```

Comment:
Yes, DeviceContext could be nullptr.
In CPU, we actually dos not need DeviceContext and we can pass a nullptr to the computing function.
For example:
```
template <typename Place, typename T>
void gemm(const CBLAS_TRANSPOSE transA, const CBLAS_TRANSPOSE transB,
          const int M, const int N, const int K, const T alpha, const T* A,
          const T* B, const T beta, T* C, platform::DeviceContext* context);
``` 
In cpu, parameter context is no used and could pass a nullptr to it.